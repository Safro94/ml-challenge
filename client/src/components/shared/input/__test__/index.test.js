import React from 'react';

import { render } from '@testing-library/react';
import { screen } from '@testing-library/dom';
import userEvent from '@testing-library/user-event';

import Input from '../';

describe('Input', () => {
	const onChange = jest.fn();
	const value = 1;

	it('should render OK', () => {
		//Act
		const { container } = render(
			<Input placeholder='placeholder' value={value} onChange={onChange} />
		);

		//Assert
		expect(container).toMatchSnapshot();
	});

	it('should call onclick', () => {
		//Act
		render(<Input onChange={onChange} value={1} />);
		userEvent.type(screen.getByRole('textbox'), 't');

		//Assert
		expect(onChange).toHaveBeenCalledTimes(1);
	});
});
