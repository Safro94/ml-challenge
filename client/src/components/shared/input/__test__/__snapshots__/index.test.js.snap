// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render OK 1`] = `
<div>
  <input
    class="input"
    placeholder="placeholder"
    type="text"
    value="1"
  />
</div>
`;
